### ISSUE TITLE
event player list in lobby may be out of sync

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#1118]
Make sure the entry list that appears as a tooltip is correct for each event.

> ts4z: weird cosmetic bug. add a new tourney, mouse over "0" players under # column, names were listed (same as previous item). reloading page fixes it. tournament must be *after* a tournament with entered players.
ts4z: (that is, when I added the 8:10 tournament, mouse over the 0; deadhead and jpmassar are listed as players in the mouse-over, but they're not actually entered

This sounds like a bug in [yew](https://yew.rs/)'s VDOM updater. Assuming I can reproduce it, I may be able to submit a useful bug report to the yew maintainers. I can probably get around the bug by simply adding a `key` attribute to each listing.

**Comment by ctm** [comment]
Oops. This is unlikely to be a bug in yew. I had forgotten that tooltips are done via `UIkit`, which is not guaranteed to work with yew. So, my guess is I simply need to add a key and always have. Luckily, the event's id makes a suitable key.

**Comment by ctm** [comment]
Deploying now.
```
commit 0c79fca7d8576efd309ffc951f27dd9cc6e5c075 (HEAD -> master)
Author: Clifford T. Matthews <ctm@devctm.com>
Date:   Sun Feb 5 06:25:32 2023 -0700

    [Bug] adds a key attribute to the tr element of an upcoming event. Closes #1118.
    
    I did not try to reproduce the bug before adding this fix.  As such,
    there's a chance that this fix doesn't work, but it's such a small bug that
    I'm willing to take that risk.

diff --git a/front-ends/web/src/upcoming/event_row.rs b/front-ends/web/src/upcoming/event_row.rs
index c45fdf9f..011a7e7a 100644
--- a/front-ends/web/src/upcoming/event_row.rs
+++ b/front-ends/web/src/upcoming/event_row.rs
@@ -363,7 +363,7 @@ impl Component for EventRow {
         };
         let join_tip = format!("{join_status}<br>{click_phrase}");
         html! {
-            <tr>
+            <tr key={event.id.to_string()}>
                 <td style={"text-align:center"}>
                     <input class={"uk-checkbox"} type={"checkbox"} {checked} uk-tooltip={join_tip} disabled={self.disabled} onclick={self.join_callback.clone()}/>
                 </td>
```
